spec/controllers/admin/instances/notes_controller_spec.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Admin::Instances::NotesController do
+  include ActionView::RecordIdentifier
+
+  render_views
+
+  let(:user) { Fabricate(:user, role: UserRole.find_by(name: 'Admin')) }
+  let(:instance_domain) { 'mastodon.example' }
+
+  before do
+    Fabricate(:account, domain: instance_domain)
+    sign_in user, scope: :user
+  end
+
+  describe 'POST #create' do
+    subject { post :create, params: params }
+
+    let(:params) { { instance_id: instance_domain, instance_note: { content: 'test content' } } }
+
+    it 'creates an instance note' do
+      expect { subject }.to change(InstanceNote, :count).by(1)
+      expect(response).to redirect_to admin_instance_path(instance_domain, anchor: dom_id(InstanceNote.last))
+    end
+
+    context 'when content is too short' do
+      let(:params) { { instance_id: instance_domain, instance_note: { content: '' } } }
+
+      it 'renders admin/instances/show' do
+        expect { subject }.to_not change(InstanceNote, :count)
+        expect(subject).to render_template 'admin/instances/show'
```

Comment:
If possible, can we assert about something on the page, rather than what view rendered? (I have a super-slow-moving effort to try to get rails-controller-testing out of project)

Reply:
Maybe, but this works for now.